docs/regular-expressions.md

```diff
@@ -0,0 +1,20 @@
+# Using Regular Expressions
+
+Regular expressions are a powerful tool. However, they are also very expensive in terms of memory. Ensuring correct and useful functionality is the priority but we have a few tips to minimize impact without affecting capabilities.
+
+* **Consider non-regular expressions options.** [`strings.Contains()`](https://pkg.go.dev/strings#Contains), [`strings.Replace()`](https://pkg.go.dev/strings#Replace), and [`strings.ReplaceAll()`](https://pkg.go.dev/strings#ReplaceAll) are dramatically faster and less memory intensive than regular expressions. If one of these will work equally well, use the non-regular expression option.
```

Comment:
Given that many (most?) of the regular expressions (at least in my experience) are used in attribute validators, maybe add some links to standard validators that can be used instead of rolling your own regex validator?

Reply:
Note above is just a suggestion for a future version of this page.
LGTM :rocket:.

Reply:
Great idea! Would be great if we had far fewer "manual" regular expressions.